engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/WithMockCompilerContext.java

```diff
@@ -0,0 +1,160 @@
+package org.enso.compiler.test.mock;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.vfs2.FileObject;
+import org.enso.compiler.Compiler;
+import org.enso.compiler.context.CompilerContext.Module;
+import org.enso.compiler.data.CompilerConfig;
+import org.enso.editions.LibraryName;
+import org.enso.pkg.Package;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import scala.jdk.javaapi.CollectionConverters;
+
+/**
+ * A {@link TestRule} to use with the {@link MockCompilerContext}.
+ *
+ * <p>I recommend to use it as a {@link org.junit.Rule}, instead of {@link org.junit.ClassRule} to
+ * ensure that the whole context is reset before each test.
+ */
+public final class WithMockCompilerContext implements TestRule {
```

Comment:
This TestRule is the entrypoint for the tests. Compiler tests are expected to use all the mocked classes only via this class (note that all other classes are package private).

Reply:
The API is still work in progress, but I believe the main idea is solid. Moreover, all the tests pass locally.